jigglypuffRL/classicalrl/sarsa/sarsa.py

```diff
@@ -0,0 +1,107 @@
+import gym
+import numpy as np
+
+class SARSA:
+    '''
+    State-Action-Reward-State-Action (SARSA)
+    Paper: http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.17.2539&rep=rep1&type=pdf
+    :param env: (Gym environment) The environment to learn from
+    :param max_iterations: (int) Maximum number of iterations per episode
+    :param max_epsiodes: (int) Maximum number of episodes
+    :param epsilon: (float) Epsilon for epsilon-greedy selection of the action
+    :param alpha: (float) Learning rate
+    :param gamma: (float) Discount Factor
+    :param lmbda: (float) Lambda for the eligibility traces
+    :param decay_rate_epsilon: (float) Decay rate for the epsilon of epsilon greedy action selection
+    :param thres_reward: (float) Threshold of the reward you want the training to stop
+    '''
+    def __init__(
+        self,
+        env,
+        max_iterations=1000,
+        max_episodes=1500001, 
+        epsilon=0.9,
+        alpha=0.1,
+        gamma=0.95,
+        lmbda=0.90,
+        decay_rate_epsilon=0.999,
+        thres_reward=0.8):
+        
+        self.env = env
+        self.max_iterations = max_iterations
+        self.max_episodes = max_episodes
+        self.epsilon = epsilon
+        self.decay_rate_epsilon = decay_rate_epsilon
+        self.alpha = alpha 
+        self.gamma = gamma
+        self.lmbda = lmbda
+        self.thres_reward = thres_reward
+        
+        # Set up the Q table 
+        self.Q_table = np.zeros((self.env.observation_space.n,self.env.action_space.n))  
```

Comment:
Should we initialize the Q_table with a random (kaiming uniform?) distribution instead of all zeros?

Reply:
Regarding this, I tried this didn't seem to train properly.

Reply:
Is it working well with all zeros? Would giving the user an option to choose initialization be a good idea? I don't know